blockchain/vm/evm.go

```diff
@@ -538,6 +538,33 @@ func (evm *EVM) CreateWithAddress(caller types.ContractRef, code []byte, gas uin
 	return evm.create(caller, codeAndHash, gas, value, contractAddr, humanReadable, codeFormat)
 }
 
+func (evm *EVM) CurrentCodeFormat() params.CodeFormat {
+	switch {
+	case evm.chainRules.IsIstanbul:
+		return params.CodeFormatEVM2
+	default:
+		return params.CodeFormatEVM
+	}
```

Comment:
Shouldn't this code be decided by the blocknumber?
```suggestion
	switch {
	case evm.chainRules.IsIstanbul:
	        switch block :=  evm.BlockNumber; {
	        case block < 50M:
	                return params.CodeFormatEVM
	        default:
	        	return params.CodeFormatEVM2
	        }
	default:
		return params.CodeFormatEVM2
	}
	return params.CodeFormatEVM2
```

Reply:
evm.chainRules.IsIstanbul does the same thing. It is set before executing according to the node's current block number. I'll add some comments at chainRules.